secret-key.js

```diff
@@ -5,9 +5,13 @@ const na = require('sodium-universal')
 // - an envelopes top level key (`msg_key`) from which others are derived
 
 module.exports = class SecretKey {
-  constructor (length) {
-    this.key = na.sodium_malloc(length || na.crypto_secretbox_KEYBYTES)
-    na.randombytes_buf(this.key)
+  constructor (lengthOrBuffer) {
+    if (Buffer.isBuffer(lengthOrBuffer)) {
+      this.key = lengthOrBuffer
+    } else {
+      this.key = na.sodium_malloc(length || na.crypto_secretbox_KEYBYTES);
```

Comment:
Aaaah, annoying that I missed this detail: `length` is not defined.

Reply:
Oh, this repo doesn't have automated tests? :) I saw no errors and assumed this must have passed the tests, or maybe there is no test of this constructor. Anyway glad you fixed it :)

Reply:
Yeah no CI tests :smiling_face_with_tear:

Reply:
SecretKey came from hmmm another module. Likely tested there. Though maybe integration test not unit test